cli/Sources/TuistGenerator/Generator/ProjectDescriptorGenerator.swift

```diff
@@ -348,3 +446,37 @@ final class ProjectDescriptorGenerator: ProjectDescriptorGenerating {
         .xcode13
     }
 }
+
+private extension PBXGroup {
+    func group(namedOrPath nameOrPath: String) -> PBXGroup? {
+        return children
+            .compactMap { $0 as? PBXGroup }
+            .first(where: { $0.name == nameOrPath || $0.path == nameOrPath })
+    }
+
+    func childReference(namedOrPath nameOrPath: String) -> PBXFileElement? {
+        return children.first(where: { $0.path == nameOrPath || ($0.name == nameOrPath) })
+    }
+}
```

Comment:
This can be moved to the body of the symbol that makes use of it. Extensions, while supported by the language and the build system, add build-time complexity.